### DISCUSSION TITLE
Charging based on sensor state

### DISCUSSION THREAD

**Discussion by djdarlo** [nathanmarlor/foxess_modbus#133] [Q&A]
I've just started using target settings on the Octopus Agile integration where you can build sensors based on the duration and energy prices. 

Is there any easy way of using this integration to charge the battery based on one of these sensors? 

I was thinking of the automation route and something like 

If target sensor is true
And SOC below a certain value
And Solar forecast for today is below a certain value
Then trigger batter charging

**Comment by canton7** [comment]
For starters, look at https://github.com/nathanmarlor/foxess_em.

If you want to do something yourself, yes you should be able to write that sort of logic using automations in home assistant. You can set min/max soc and work mode, and the next release will let you set charge windows as well.

**Comment by djdarlo** [comment]
Thank you. I’m thinking the force charge on and off services may do the
trick.

**Comment by imvikysharma** [comment]
Hi there! Yes, it's definitely possible to use the Octopus Agile integration to charge your battery based on a target sensor. The automation route you've proposed seems like a good starting point.

Assuming you have a battery charger that can be controlled via an API or smart plug, you could use an automation platform like Home Assistant or Node-RED to create a rule that triggers the charger when the conditions you've specified are met.

For example, you could use a Home Assistant automation with the following trigger and conditions:

Trigger: Target sensor changes to "true"
Conditions:

Battery SOC is below a certain value
Solar forecast for today is below a certain value
Action: Turn on the battery charger

I would like to recommend a book on Agile Software Development.

https://www.amazon.com/dp/B0BVRW597Q?maas=maas_adg_A4F7B0AEC381971BCB2D877AFF231AAB_afap_abs&ref_=aa_maas&tag=maas
